test/unstoppable/AbstractAttacker.sol

```diff
@@ -6,26 +6,15 @@ import "../../lib/halmos-cheatcodes/src/SymTest.sol";
 import "forge-std/Test.sol";
 
 contract AbstractAttacker is Test, SymTest {
-    mapping (uint256 => address) known_addresses;
-    uint256 known_next_index = 0;
+	function attack(address[] calldata targets) public {
+        address target = svm.createAddress("target");
 
-    function add_known_address(address known) public {
-        known_addresses[known_next_index] = known;
-        known_next_index++;
-    }
-
-	function attack() public {
-        uint256 contract_index = svm.createUint256('contract_index');
-        vm.assume(contract_index < known_next_index);
-        for (uint256 i = 0; i < known_next_index; i++) { // brootforce is happening here
-            if (i == contract_index) {
-                address to_call = known_addresses[i]; // contract address is not symbolic now
+        for (uint i = 0; i < targets.length; i++) {
+            if (target == targets[i]) {
```

Comment:
this will produce a more readable output, and we avoid touching storage

Reply:
I agree that this will produce a more readable output and looks more intuitive. But I decided to use a map-based solution because of the next reason:
In the future, there is a challenge called "backdoor", that requires creating of a new contract and calling it. Addresses, passed as a function parameter are constant. So I want to make more "universal" skeleton for future solutions, based on this solution.
By the way, what do you think about the next idea: Each contract address in Halmos is "000...aaaaxxxx". and these xxxx are going one-after-one. Can we iterate over these addresses to achieve the same result?

Reply:
> Addresses, passed as a function parameter are constant. So I want to make more "universal" skeleton for future solutions, based on this solution.

I don't understand this objection -- in the end we're still just passing a dynamic list of known addresses. 

`attacker.add_known_address(address(token))` becomes `known_addresses[0] = address(token)`, what's more constant about it?

Reply:
> By the way, what do you think about the next idea: Each contract address in Halmos is "000...aaaaxxxx". and these xxxx are going one-after-one. Can we iterate over these addresses to achieve the same result?

we technically could, but I wouldn't recommend it as it may change in the future. Contracts created with CREATE2 are of the form `"000...bbbbxxxx"`

Reply:
> > Addresses, passed as a function parameter are constant. So I want to make more "universal" skeleton for future solutions, based on this solution.
> 
> I don't understand this objection -- in the end we're still just passing a dynamic list of known addresses.
> 
> `attacker.add_known_address(address(token))` becomes `known_addresses[0] = address(token)`, what's more constant about it?

I want to handle the case, when the list of contracts could be changed during "attack()" execution. As far as I know, when we pass address array as a parameter, we can't add another address to that list